messaging/cloud-events/amqp-binary/src/main/java/io/quarkus/ts/messaging/amqpreactive/Consumer.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.ts.messaging.amqpreactive;
+
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.jboss.logging.Logger;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.ce.impl.DefaultIncomingCloudEventMetadata;
+
+@ApplicationScoped
+public class Consumer {
+
+    private static final Logger LOG = Logger.getLogger(Producer.class.getName());
+
+    private final AtomicReference<HashMap> results = new AtomicReference<>();
+
+    @Incoming("cloud-events")
+    public Uni<Void> process(Message message) {
+        Integer price = Integer.valueOf(message.getPayload().toString());
+        LOG.info("process fired: " + price);
+
+        DefaultIncomingCloudEventMetadata metadata = retrieveCloudEventMetaData(message);
+
+        HashMap<String, String> result = new HashMap<>();
+        result.put("source", metadata.getSource().toString());
+        result.put("subject", metadata.getSubject().orElse(null).toString());
```

Comment:
With the current implementation, did you try to see if `.orElse(null).toString()` give you a null pointer exception if `metadata.getSubject()` is null? 

Instead of `.orElse(null)` what about return an empty String.